Bottom: 66522de14c2912c83e6499db689a35f398c43a9c
Top:    a00a46543c9c1a37bb586b19c01a0d8ab1d80891
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-08 09:35:32 +0800

Refresh of autofs-5.0.7-fix-potential-null-dereference-in-lookup_mount.patch

---

diff --git a/modules/lookup_yp.c b/modules/lookup_yp.c
index f6095ad..d70605a 100644
--- a/modules/lookup_yp.c
+++ b/modules/lookup_yp.c
@@ -698,18 +698,8 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 		ret = ctxt->parse->parse_mount(ap, key, key_len,
 					       mapent, ctxt->parse->context);
 		if (ret) {
-			time_t now = time(NULL);
-			int rv = CHE_OK;
-
 			cache_writelock(mc);
-			me = cache_lookup_distinct(mc, key);
-			if (!me)
-				rv = cache_update(mc, source, key, NULL, now);
-			if (rv != CHE_FAIL) {
-				me = cache_lookup_distinct(mc, key);
-				if (me)
-					me->status = now + ap->negative_timeout;
-			}
+			cache_update_negative(mc, source, key, ap->negative_timeout);
 			cache_unlock(mc);
 		}
 	 }
